src/tests/class_tests/openms/source/FASTAFile_test.cpp

```diff
@@ -53,141 +53,187 @@ START_TEST(FASTAFile, "$Id$")
+                FASTAFile file;
+
+                file.load(OPENMS_GET_TEST_DATA_PATH("FASTAFile_test.fasta"),data);
+                TEST_EXCEPTION(Exception::UnableToCreateFile, file.store("/bla/bluff/blblb/sdfhsdjf/test.txt",data))
+
+                file.store(tmp_filename,data);
+                file.load(tmp_filename,data2);//fail
+                TEST_EQUAL(data==data2,true);
+        END_SECTION
+
+
+        START_SECTION([EXTRA] test_strange_symbols_in_sequence)
+                // test if * is read correctly (not changed into something weird like 'X')
+                String tmp_filename;
+                NEW_TMP_FILE(tmp_filename);
+                FASTAFile file;
+                vector<FASTAFile::FASTAEntry> data, data2;
+                FASTAFile::FASTAEntry temp_entry;
+                temp_entry.identifier = String("P68509|1433F_BOVIN");
+                temp_entry.description = String("This is the description of the first protein");
+                temp_entry.sequence = String("GDREQLLQRAR*LAEQ*AERYDDMASAMKAVTEL");
+                data.push_back(temp_entry);
+                data.push_back(temp_entry); // twice
+
+                file.store(tmp_filename, data);
+                file.load(tmp_filename, data2);
+
+                ABORT_IF(data2.size() != 2);
+                TEST_EQUAL(data2[0] == temp_entry, true);
+                TEST_EQUAL(data2[1] == temp_entry, true);
+
+        END_SECTION
+
+        START_SECTION([EXTRA] test_white_spaces) // noch eine Sequenz in die Fasta Datei hinzufügen und da checken, ob die Zeichen entfernt werden
+                // test if ' ' and '\t' are removed correctly
+                String tmp_filename;
+                NEW_TMP_FILE(tmp_filename);
+                FASTAFile file;
+                vector<FASTAFile::FASTAEntry> data, data2;
+                FASTAFile::FASTAEntry temp_entry;
+                temp_entry.identifier = String("P68509|1433F_BOVIN");
+                temp_entry.description = String("This is the description of the first protein");
+                temp_entry.sequence = String("GDREQLLQRAR LAEQ\tAERYDDMASAMKAVTEL");
+                data.push_back(temp_entry);
+
+                file.store(tmp_filename, data);//tmp_filename ist dann outfile, den wir gleich in load übergeben
+                file.load(tmp_filename, data2);
+
+                ABORT_IF(data2.size() != 1);
+                TEST_EQUAL(data2[0].sequence == string("GDREQLLQRARLAEQAERYDDMASAMKAVTEL"), true);
+
+        END_SECTION
+
+        START_SECTION([EXTRA] test_position) //
+                // test if setPosition() works correctly
+                String tmp_filename;
+                NEW_TMP_FILE(tmp_filename);
+                FASTAFile file;
+                vector<FASTAFile::FASTAEntry> data;
+                FASTAFile::FASTAEntry temp_entry;
+                file.readStart(OPENMS_GET_TEST_DATA_PATH("FASTAFile_test.fasta"));
+                streampos a = file.position();
```

Comment:
you could make the test a bit more exhaustive by testing the following:
read the first, say, 4 entries and remember their FASTAEntry's as well as their position.
Then reset the position to the, say, 2nd entry and read the next 3 entries, and remember the position.
Now compare if the positions and entries for the things that were read twice match.